internal/registration/registration.go

```diff
@@ -87,3 +96,26 @@ func (r *Registration) registerDeviceOnce() error {
 	}
 	return nil
 }
+
+func (r *Registration) Deregister(configuration models.DeviceConfigurationMessage) error {
+	err := r.workloads.Deregister()
```

Comment:
For the `syncPaths()` we can return here the list of the stopped workloads and provide them to `r.monitor.Deregister()`
which will shutdown the ticker and will call synchronously the `syncPaths()` with the given list of workloads that need to be synced.

We can even to monitor.StopSyncPaths() (new function) when this function starts to stop the ticker, then continue with the steps above.

In this scenario, assuming I'm not missing anything, a single syncPaths(workloads) is enough.